test/subscribers/publish_zync_event_subscriber_test.rb

```diff
@@ -1,14 +1,67 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class PublishZyncEventSubscriberTest < ActiveSupport::TestCase
-  def setup
-    @subscriber = PublishZyncEventSubscriber.new
+
+  class OIDCApplicationEventTest < ActiveSupport::TestCase
+    attr_reader :event, :subscriber, :publisher
+
+    setup do
+      service = FactoryBot.create(:simple_service, backend_version: 'oauth')
+      application = FactoryBot.create(:simple_cinstance, service: service)
+      @event = Applications::ApplicationCreatedEvent.create(application, nil)
+      @publisher = mock('publisher')
+      @subscriber = PublishZyncEventSubscriber.new(publisher)
+    end
+
+    test 'publish Zync Event for OIDC auth always' do
+      publisher.expects(:call).times(3).returns(:ok)
```

Comment:
If you use a mock, you can use `assert` the mock returns `:ok`. Then, IMO the `expects` is redundant.

Reply:
I also want to verify it's called N times. What is the suggested syntax though and how it compares to `expects`?...

Reply:
What I suggest is not using `expects`, just `assert_equal :ok, @subscriber.call(event)` or even the current `assert @subscriber.call(event)`. But if you want to verify the number of calls, then you need to use `expects`